File: cpp/kode program/Food.cpp
#include "MenuItem.cpp"

// kelas food adalah turunan dari menuitem
class Food : public MenuItem {
private:
    int spicyLevel;   // tingkat kepedasan makanan
    int calories;     // jumlah kalori makanan

public:
    Food() {}

    // setter dan getter untuk spicy level
    void setSpicyLevel(int spicyLevel) { 
        this->spicyLevel = spicyLevel; 
    }
    int getSpicyLevel() { 
        return spicyLevel; 
    }

    // setter dan getter untuk calories
    void setCalories(int calories) { 
        this->calories = calories; 
    }
    int getCalories() { 
        return calories; 
    }

    // override method printinfo dari parent
    // menampilkan informasi lengkap makanan termasuk spicy level dan calories
    void printInfo() override {
        MenuItem::printInfo();
        cout << " | Spicy Level: " << spicyLevel
             << " | Calories: " << calories << endl;
    }
};

File: python/kode program/Food.py
from MenuItem import MenuItem

class Food(MenuItem):
    def __init__(self, item_id=0, name="", price=0, spicy_level=0, calories=0):
        super().__init__(item_id, name, price)  # panggil constructor menuitem
        self._spicy_level = spicy_level  # atribut private spicy_level
        self._calories = calories        # atribut private calories

    # setter dan getter untuk spicy_level
    def set_spicy_level(self, spicy_level):
        self._spicy_level = spicy_level

    def get_spicy_level(self):
        return self._spicy_level

    # setter dan getter untuk calories
    def set_calories(self, calories):
        self._calories = calories

    def get_calories(self):
        return self._calories

    # override print_info dari class induk
    def print_info(self):
        info = self.base_info()
        info += f" | spicy level: {self._spicy_level} | calories: {self._calories}"
        print(info)
